packages/loopring_v3/contracts/aux/agents/FastWithdrawalAgent.sol

```diff
@@ -44,119 +38,110 @@ import "../../lib/SignatureUtil.sol";
 ///        authorize this contract as their agent.
 ///
 /// @author Brecht Devos - <brecht@loopring.org>
+/// @author Kongliang Zhong - <kongliang@loopring.org>
+/// @author Daniel Wang - <daniel@loopring.org>
 contract FastWithdrawalAgent is ReentrancyGuard
 {
     using AddressUtil       for address;
     using AddressUtil       for address payable;
     using ERC20SafeTransfer for address;
     using MathUint          for uint;
-    using SignatureUtil     for bytes32;
 
-    struct FastWithdrawal
+    event Processed(
+        address exchange,
+        address from,
+        address to,
+        address token,
+        uint96  amount,
+        address provider,
+        bool    success
+    );
+
+    struct Withdrawal
     {
         address exchange;
         address from;                   // The owner of the account
-        address to;                     // The address that will receive the tokens withdrawn
+        address to;                     // The `to` address of the withdrawal
         address token;
         uint96  amount;
-        address feeToken;
-        uint96  fee;
-        uint32  nonce;
-        uint32  validUntil;
-
-        bytes   signature;
+        uint32  storageID;
     }
 
-    // EIP712
-    bytes32 constant public FASTWITHDRAWAL_TYPEHASH = keccak256(
-        "FastWithdrawal(address exchange,address from,address to,address token,uint96 amount,address feeToken,uint96 fee,uint32 nonce,uint32 validUntil)"
-    );
-    bytes32 public DOMAIN_SEPARATOR;
-
-    constructor()
+    // This method needs to be called by any liquidity provider
+    function execute(Withdrawal[] calldata withdrawals)
+        public
+        nonReentrant
+        payable
     {
-        DOMAIN_SEPARATOR = EIP712.hash(EIP712.Domain("FastWithdrawalAgent", "1.0", address(this)));
+        // Do all fast withdrawals
+        for (uint i = 0; i < withdrawals.length; i++) {
+            executeInternal(withdrawals[i]);
+        }
+        // Return any ETH left into this contract
+        // (can happen when more ETH is sent than needed for the fast withdrawals)
+        msg.sender.sendETHAndVerify(address(this).balance, gasleft());
     }
 
     // This method needs to be called by any liquidity provider
-    function executeFastWithdrawals(FastWithdrawal[] memory fastWithdrawals)
+    function execute(Withdrawal calldata withdrawal)
```

Comment:
we may want to remove this method, as in FastWithdrawalProvider, only `execute(Withdrawal[] calldata withdrawals)` is used.

Reply:
maybe also change the name to something other than `execute` to differentiate it from the function in FastWithdrawalProvider.